chore : 디자인 시스템 구축 및 레이아웃 설정

KimDongGyun23 opened the pull request:
## 변경 사항

- 컬러 팔레트 설정
- 폰트 설정
- 그림자 설정
- flex 공용 레이아웃 설정
- 화면 레이아웃 설정

<br/>

## 리뷰 필요

피그마 기반으로 디자인 시스템 구축하였습니다. 
변수명은 혼란을 막기 위해 피그마에 적힌 그대로 작성하였습니다.

<br/>

### 컬러
기존 tailwind css에 덮어씌우려 했으나, 등록되지 않은 색상이 사용되는 것으로 확인되어, 확장성을 붙였습니다.

<br/>

### 폰트
`className = "subtitle-B"` 와 같이 `"소문자 이름-굵기"` 로 작성하시면 됩니다.
utilities에 작성하여 자동완성이 되지않으니 주의해주세요.

<br/>

### flex 레이아웃
자주 사용하는 flex 레이아웃들을 클래스로 생성하였습니다.
이 또한, 자동완성이 되지않으니 주의해주세요.

<br/>

### 화면 레이아웃 설정
최소 화면 너비를 320px, 최대 화면 너비를 450px로 설정하였습니다.
웹에서 보이는 화면을 정중앙 배치하였습니다.
현재 가장자리 테두리를 설정하였는데, 개발을 위한 용도이고 완성 후 삭제 예정입니다.

<br/><br/>

### 첨부 링크(옵션)
(breackpoint)
https://velog.io/@sangpok/%EB%B0%98%EC%9D%91%ED%98%95-UI-%ED%95%B4%EC%83%81%EB%8F%84-%EA%B8%B0%EC%A4%80

close #13

KimDongGyun23 committed:
chore : 컬러 팔레트 설정

컬러 팔레트 설정
globals.css 내부 tailwind 속성 분리
#13

KimDongGyun23 committed:
chore : NotoSansKR 폰트 추가
#13

KimDongGyun23 committed:
chore : 전역 폰트 설정
#13

KimDongGyun23 committed:
chore : 폰트 utilities 설정
#13

KimDongGyun23 committed:
chore : tailwind.config 폰트 설정

폰트 굵기 400, 500, 700으로 고정
폰트 크기 및 행간높이 추가

#13

KimDongGyun23 committed:
chore : 그림자 속성 추가
#13

KimDongGyun23 committed:
chore : 공용 스타일 추가

flex를 이용한 레이아웃들 추가

#13

KimDongGyun23 committed:
chore : 화면 크기 설정

모바일 기준 최소 최대 너비 설정
웹 기준 가운데 정렬

#13

KimDongGyun23 committed:
fix : letter-space 수정

%가 설정되지 않아 px 단위로 변경

#13

kimsuyeon0916 commented on tailwind.config.ts:
color, fontSize, boxShadow 등 을 ts 파일로 분리해서 styles 폴더에 보관하고 import 해오는 거 어떻게 생각하시나용??🤔 동균님 의견이 궁금합니다!

KimDongGyun23 replied:
저도 그 방식을 생각했었는데 모듈을 찾지 못한다는 오류에 마주치게 되었습니다.
해당 오류를 해결하지 못하여 위와 같은 방식을 선택했던 것 같습니다.

오늘 중으로 다시 해결해보고 올리도록 하겠습니다. 감사합니다 :)

KimDongGyun23 replied:
해당 부분 수정해서 올렸습니다.

kimsuyeon0916 commented on public/fonts/NotoSansKR-Bold.ttf:
google 폰트에서 notosansKR 폰트가 제공되는걸로 아는데, 혹시 따로 추가하신 이유가 궁금합니다!
성능 최적화를 위해 아래의 방법대로 설정하는건 어떤지, 동균님의 의견이 궁금합니다!

https://nextjs.org/docs/app/building-your-application/optimizing/fonts#google-fonts

KimDongGyun23 replied:
네 한번 알아보고 수정해두겠습니다. 감사합니다.

KimDongGyun23 replied:
해당 부분 수정해서 올렸습니다.

KimDongGyun23 committed:
chore : tailwind 커스텀 속성 파일 분리

color, fontSize, lineHeight, boxShadow, backGroundImage 각각의 파일로 분리하여 import 처리

#13

KimDongGyun23 committed:
chore : 폰트 설정 변경

폰트를 추가하는 기존의 방식에서 구글 폰트를 자동으로 가져오는 방식으로 변경

#13

kimsuyeon0916 commented on tailwind.config.ts:
```diff
@@ -8,11 +15,14 @@ const config: Config = {
-        'gradient-radial': 'radial-gradient(var(--tw-gradient-stops))',
-        'gradient-conic': 'conic-gradient(from 180deg at 50% 50%, var(--tw-gradient-stops))',
-      },
+      backgroundImage: { ...BACKGROUND_IMAGE },
+      backgroundColor: { ...COLORS },
+      colors: { ...COLORS },
+      fontSize: { ...FONT_SIZE },
+      lineHeight: { ...LINE_HEIGHT },
+      boxShadow: { ...BOX_SHADOW },
```
해당 요소들을 tailwindcss Configuration 과 동일한 네이밍컨벤션(카멜케이스)를 사용해서  아래와 같이 작성하는것도 좋을 것 같습니다!

```suggestion
      extend: {
        fontWeight,
        backgroundImage,
        backgroundColor
        colors,
        fontSize,
        lineHeight,
        boxShadow
    }
```

KimDongGyun23 replied:
네. 해당 사항 반영해서 다시 올렸습니다.

kimsuyeon0916 commented on src/app/layout.tsx:
```diff
@@ -1,15 +1,22 @@
 
 import './globals.css'
 
+const notoSans = Noto_Sans_KR({ subsets: ['latin'], weight: ['400', '500', '700'] })
+
```
```suggestion
const notoSans = Noto_Sans_KR({
  subsets: ['latin'],
  variable: '--font-notoSansKR',
})
```

가변 글꼴이라, 따로 fontWeight 지정 안해줘도 됩니다! 
tailwindCss 와 함께 사용할때는 classname 대신 variables 로 공식문서에서는 지정하네요!

https://nextjs.org/docs/app/building-your-application/optimizing/fonts#with-tailwind-css

KimDongGyun23 replied:
네. 해당 사항 반영해서 다시 올렸습니다.

KimDongGyun23 replied:
폰트 굵기를 ["400", "500", "700"]으로 고정하고싶었는데, 해당 방식이 아닌 tailwind에서 지정하는 것이군요.

수정해서 올렸습니다.

KimDongGyun23 committed:
chore : tailwind css 속성 내보내기 변수명 변경

LINE_HEIGHT => lineHeight
tailwindcss와 동일한 네이밍 컨벤션으로 변경

#13

KimDongGyun23 committed:
chore : style 배럴파일 생성

tailwind 커스텀 속성 배럴파일로 내보낸 후, tailwind.config에서 한번에 import

#13

KimDongGyun23 committed:
chore : 글꼴 방식 변경

가중치 삭제, variable을 통하여 tailwind에 맞도록 수정

#13

kimsuyeon0916 commented on src/app/layout.tsx:
```diff
@@ -1,15 +1,26 @@
 export default function RootLayout({ children }: PropsWithChildren) {
   return (
-    <html lang="en">
+    <html lang="en" className={notoSansKR.className}>
```
lang="ko" 로 바꿔놓는것도 좋을 것 같습니다!

KimDongGyun23 replied:
넵 변경했습니다. :)

kimsuyeon0916 commented:
동균님 정말 고생 많으셨습니다! 이제 저희 한번 달려볼까요옹~~🚀

KimDongGyun23 committed:
chore : 언어 "en"에서 "ko" 로 수정

#13
